com/course/zenkouv/src/test/java/calculator/CalculatorTest.java

```diff
@@ -0,0 +1,74 @@
+package calculator;
+
+import lecture5.util.operation.Operation;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class CalculatorTest {
+
+    private Operation operation;
+
+    @Test
+    public void createOperation() {
+        operation = new Operation();
+        Assert.assertNotNull(operation);
+    }
+
+    @Test(dataProvider = "additionOperationData", dependsOnMethods = "createOperation")
+    public void testAdditionOperation(double variable1, double variable2, double expectedResult) {
+        Assert.assertEquals(operation.addition(variable1, variable2), expectedResult);
+    }
+
+    @Test(dataProvider = "subtractionOperationData", dependsOnMethods = "createOperation")
+    public void testSubtractionOperation(double variable1, double variable2, double expectedResult) throws InterruptedException {
+        Thread.sleep(10000);
+        Assert.assertEquals(operation.subtraction(variable1, variable2), expectedResult);
+    }
+
+    @Test(dataProvider = "divisionOperationData", dependsOnMethods = "createOperation")
+    public void testDivisionOperation(double variable1, double variable2, double expectedResult) {
```

Comment:
it is bad practice to mix positive and negative cases in one test
tests should be as smaller as it possible
so write two tests for division operation

Reply:
fix, add separate test for exception